test/integration_test.dart

```diff
@@ -418,6 +418,34 @@ defineTests() {
       });
     });
 
+    group('always_require_non_null_named_parameters', () {
+      IOSink currentOut = outSink;
+      CollectingSink collectingOut = new CollectingSink();
+      setUp(() {
+        exitCode = 0;
+        outSink = collectingOut;
+      });
+      tearDown(() {
+        collectingOut.buffer.clear();
+        outSink = currentOut;
+        exitCode = 0;
+      });
+
+      test('only throw errors', () {
+        dartlint.main([
+          'test/_data/always_require_non_null_named_parameters',
+          '--rules=always_require_non_null_named_parameters'
```

Comment:
> Can we pass the flag in here?

It fails with the message:

>Could not find an option named "assert-initializer".

Your last proposal (_add support to linter main to allow you to specify analysis options_) seems to be the only solution. But it's a change that need to be done in the analyzer package not here.